### PR TITLE
WT-2402 Misaligned structure accesses lead to undefined behavior

### PR DESCRIPTION
**Pull request by keithbostic**
@michaelcahill, @agorrod, @daveh86: I ended up turning off `std=c11` on Solaris, everything else was just too painful, and involved setting compiler flags that weren't strictly documented.

I also made a pass through `configure.ac.in` to try and clean up how we were setting flags as well as the order in which we set them.

I haven't yet tested this as much as I'd like, but I wanted to push it so you folks have a chance to see where I'm heading and say "no".

### CODE REVIEW

**Comment by keithbostic** [comment]
@sueloverso, c30aed7 is some lint fixes for the new `alter()` function in `format`, can you please take a fast look?

It wasn't clear to me why we ignore all errors from `WT_SESSION.alter`, that might be worth a comment.

**Comment by sueloverso** [comment]
Those changes look fine.  If we get an error, the thread exits.

File: build_posix/configure.ac.in [hunk 1]
```diff
@@ -99,11 +83,35 @@ if test "$GCC" = "yes"; then
...
+		AM_GCC_WARNINGS($wt_cv_cc_version);;
+	*)
+		AC_MSG_ERROR(
+		    [--enable-strict does not support "$wt_cv_cc_version".]);;
+	esac
+
+	AM_CFLAGS="$AM_CFLAGS $wt_cv_strict_warnings"
+fi
+
+# Configure options.
```
Comment (agorrod) [inline build_posix/configure.ac.in]: Why did you move ``AM_OPTIONS`` lower?

Comment (keithbostic) [inline build_posix/configure.ac.in]: > Why did you move `AM_OPTIONS` lower?

Because I wanted to get the "principle" compiler flags in place before doing anything else.

For example, previously, we called `AM_OPTIONS` before setting `-pthreads` or defining `_GNU_SOURCE`, and if `AM_OPTIONS` does tests on the system (which seems reasonable), they won't test using the same compiler environment the build will have.

**Comment by keithbostic** [comment]
This change seems to work everywhere, so I'm going to merge it.
